src/components/shared/LessonSide/LevelSelector.tsx

```diff
@@ -1,5 +1,48 @@
-function LevelSelector(): JSX.Element {
-  return <div>Level Selector</div>;
+import '../../../styles/LessonSide.scss';
+
+interface LevelSelectorProps {
+  numLevels: number;
+  currentLevel: number;
+  maxLevelReached: number;
+}
+
+function LevelSelector({
+  numLevels,
+  currentLevel,
+  maxLevelReached,
+}: LevelSelectorProps): JSX.Element {
+  return (
+    <div className="selector-box">
+      <div className="level-selector">
+        {
+          //numLevels * 2 - 1 elements in Array as one button for each layer and numLevels - 1 connectors.
```

Comment:
We should add this to linting, but generally, spaced comments are the standard
https://eslint.org/docs/rules/spaced-comment
```suggestion
          // numLevels * 2 - 1 elements in Array as one button for each layer and numLevels - 1 connectors.
```